Makefile

```diff
@@ -39,16 +39,30 @@ K8S_YAMLS_LOCATION_STAGING=./k8s/staging
 export CONTAINER_REGISTRY := gcr.io/flow-container-registry
 export DOCKER_BUILDKIT := 1
 
-# setup the crypto package under the GOPATH: needed to test packages importing flow-go/crypto
-.PHONY: crypto_setup_gopath
-crypto_setup_gopath:
-	bash crypto_setup.sh
+# `ADX_SUPPORT` is 1 if ADX instructions are supported and 0 otherwise.
```

Comment:
ADX leads to better performance and not all CI/build machines have ADX. 
Default is to use ADX unless the machine doesn't support it.

Reply:
looks like this block of code is repeated in all of the makefiles. Could we move it into a shared script and source it?

Reply:
To encapsulate this into a reusable file that's references by multiple makefiles:
- create `adx.mk` and add the ADX checking code there
- reference `adx.mk` from `Makefile` like this:

`include adx.mk`

I tested this locally with `Makefile` and it works:

```
# docker container registry
export CONTAINER_REGISTRY := gcr.io/flow-container-registry
export DOCKER_BUILDKIT := 1

include adx.mk

cmd/collection/collection:
	$(CGO_FLAG) go build -o cmd/collection/collection cmd/collection/main.go
```


Output:

```
make cmd/collection/collection
CGO_CFLAGS="" go build -o cmd/collection/collection cmd/collection/main.go
```

Reply:
Great suggestion from you both!

I replaced the logic by the `.mk` file in 3 Makefiles. I left `crypto/Makefile` unchanged as the crypto module is soon going to be moved to a separate repo `onflow/crypto` and can't share the same `.mk` file.

https://github.com/onflow/flow-go/pull/4817/commits/f64d5ea4fdd97372e35cde00d37092734e2bb3f3